src/PointwiseFunctions/Hydro/StressEnergy.cpp

```diff
@@ -70,23 +99,78 @@ void stress_trace(gsl::not_null<Scalar<DataType>*> result,
           (square(get(lorentz_factor)) * get(spatial_velocity_squared) + 1.);
 }
 
+template <typename DataType>
+void stress_energy_tensor(
+    gsl::not_null<tnsr::AA<DataType, 3>*> result,
+    const Scalar<DataType>& rest_mass_density,
+    const Scalar<DataType>& specific_internal_energy,
+    const Scalar<DataType>& pressure, const Scalar<DataType>& lorentz_factor,
+    const Scalar<DataType>& lapse,
+    const Scalar<DataType>& comoving_magnetic_field_magnitude,
+    const tnsr::I<DataType, 3>& spatial_velocity,
+    const tnsr::I<DataType, 3>& shift,
+    const tnsr::I<DataType, 3>& magnetic_field,
+    const tnsr::ii<DataType, 3>& spatial_metric,
+    const tnsr::II<DataType, 3>& inverse_spatial_metric) {
+  const auto inverse_spacetime_metric =
+      gr::inverse_spacetime_metric(lapse, shift, inverse_spatial_metric);
+
+  const auto magnetic_field_dot_spatial_velocity =
+      dot_product(magnetic_field, spatial_velocity, spatial_metric);
+
+  const auto comoving_magnetic_field_v = comoving_magnetic_field(
+      spatial_velocity, magnetic_field, magnetic_field_dot_spatial_velocity,
+      lorentz_factor, shift, lapse);
+
+  const auto four_velocity_v =
+      four_velocity(spatial_velocity, shift, lorentz_factor, lapse);
+
+  const auto rho_h_star =
+      (get(rest_mass_density) +
+       get(rest_mass_density) * get(specific_internal_energy)) +
+      get(pressure) + square(get(comoving_magnetic_field_magnitude));
+
+  const auto p_star =
+      get(pressure) + square(get(comoving_magnetic_field_magnitude)) / 2.;
```

Comment:
You don't want auto here because this creates an expression tree that gets evaluated many times in the loop below. Specify the type